opendrift/export/io_netcdf.py

```diff
@@ -3,235 +3,82 @@
+                     if n not in self.elements_scheduled.ID-1]
+        self.result = self.result.isel(trajectory=seeded_indices)
 
     # Finally changing UNLIMITED time dimension to fixed, for CDM compliance.
-    # Fortunately this is quite fast.
-    # https://www.unidata.ucar.edu/software/thredds/current/netcdf-java/reference/FeatureDatasets/CFpointImplement.html
-    try:
-        logger.debug('Making netCDF file CDM compliant with fixed dimensions')
-        if self.num_elements_scheduled() > 0:
-            logger.info('Removing %i unseeded elements already written to file' % self.num_elements_scheduled())
-            mask = np.ones(self.history.shape[0], dtype=bool)
-            mask[self.elements_scheduled.ID-1] = False
-        with Dataset(self.outfile_name) as src, \
-                Dataset(self.outfile_name + '_tmp', 'w') as dst:
-            for name, dimension in src.dimensions.items():
-                if name=='trajectory':
-                    # Truncate dimension length to  number actually seeded
-                    dst.createDimension(name, self.num_elements_activated())
-                else:
-                    dst.createDimension(name, len(dimension))
-
-            for name, variable in src.variables.items():
-                if '_FillValue' in variable.ncattrs():
-                    fill = variable.getncattr('_FillValue')
-                else:
-                    fill = None
-                dstVar = dst.createVariable(name, variable.datatype,
-                                             variable.dimensions, fill_value=fill)
-                srcVar = src.variables[name]
-                # Truncate data to number actually seeded
-                if 'trajectory' in variable.dimensions:
-                    if self.num_elements_scheduled() > 0:
-                        if len(variable.dimensions) == 2:
-                            dstVar[:] = srcVar[mask, :]
-                        else:
-                            dstVar[:] = srcVar[mask]  # Copy data
-                    else:
-                        dstVar[:] = srcVar[:]
-                else:
-                    dstVar[:] = srcVar[:]
-                for att in src.variables[name].ncattrs():
-                    # Copy variable attributes
-                    if att in ['_FillValue']:
-                        continue
-                    dstVar.setncattr(att, srcVar.getncattr(att))
-
-            for att in src.ncattrs():  # Copy global attributes
-                dst.setncattr(att, src.getncattr(att))
-
-        move(self.outfile_name + '_tmp', self.outfile_name)  # Replace original
-    except Exception as me:
-        print(me)
-        print('Could not convert netCDF file from unlimited to fixed dimension. Could be due to netCDF library incompatibility(?)')
+    logger.debug('Making netCDF file CDM compliant with fixed dimensions')
+    self.result.to_netcdf(self.outfile_name + '_tmp', unlimited_dims={})
```

Comment:
What if _tmp exists. Better to use python tempfile.

Reply:
Yes, will do